### PR TITLE
Switched from JSHint to ESLint

### PR DESCRIPTION
**Pull request by ficristo**
Similar work to the one in Brackets.
I've added a `.bracket.json` file so I could change the prefered linter. (And remove the inline jslint declarations)
I've added an `.eslint.json` file under `tasks/` because Grunt isn't compatible with the rule `no-invalid-this`.
There should not be any change in behaviour.

### CODE REVIEW

File: appshell/node-core/DomainManager.js [hunk 1]
```diff
@@ -21,15 +21,10 @@
...
 
-/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4,
-maxerr: 50, node: true */
-/*global */
-
 (function () {
     "use strict";
         
     var util              = require("util"),
-        server            = require("./Server"),
```
Comment (zaggino) [inline appshell/node-core/DomainManager.js]: why this change?

File: tasks/set-release.js [hunk 2]
```diff
@@ -48,7 +47,6 @@ module.exports = function (grunt) {
             packageJSON                 = grunt.file.readJSON(packageJsonPath),
             winInstallerBuildXmlPath    = "installer/win/brackets-win-install-build.xml",
             buildInstallerScriptPath    = "installer/mac/buildInstaller.sh",
-            wxsPath                     = "installer/win/Brackets.wxs",
```
Comment (zaggino) [inline tasks/set-release.js]: is this a valid change?

**Comment by ficristo** [comment]
@zaggino They are both unused.

**Comment by zaggino** [comment]
Yep, but even requiring them can have side-effects, like parts of the module being executed, event listeners attached and so on. Are you sure not requiring them won't break anything? Instead of `server            = require("./Server")` you can do just `require("./Server")` to be sure nothing breaks.

**Comment by ingorichter** [comment]
What is the status of this? Any progress? @ficristo @zaggino.

**Comment by zaggino** [comment]
Seems finished to me

**Comment by ficristo** [comment]
I've squashed the commit and enabled the `no-trailing-spaces` rule too.
Should be ready.

**Comment by nethip** [comment]
@ficristo I tried playing with this branch. There is one side effect because of this change. After building brackets-shell with this branch and opening brackets-shell source folder, with the built shell, JS hints don't appear anymore.

screen shot 2016-06-21 at 10 54 39 pm

The below snapshot is with the shell built from the release branch.

screen shot 2016-06-21 at 10 55 24 pm

Also I have observed that if I open any file under node-core with any other project, JSHints kick in.
screen shot 2016-06-21 at 10 56 45 pm

**Comment by ficristo** [comment]
I cannot replicate the issue on the node-core side.
As of now there is a newer version of ESLint which will rise some more warnings and would like to update to it.
But before can we get a consensus on what to do with these:
https://github.com/adobe/brackets/issues/11984
https://github.com/adobe/brackets/pull/11988
(For not seeing JSLint anymore: we need the ESLint extension in core or I need to reenable JSLint like I did on Brackets side)

**Comment by ficristo** [comment]
I've reintroduced JSLint so we can merge this.
I removed the IIFE and so deindented the code, that's why you see a lot of changes.
You can look at this link https://github.com/adobe/brackets-shell/pull/542/files?w=1 to see code changes only.

**Comment by zaggino** [comment]
Gone through it, LGTM
